src/sentry/static/sentry/app/views/organizationIntegrations/integrationStatus.tsx

```diff
@@ -22,12 +22,12 @@ const IntegrationStatus = styled((props: StatusProps) => {
   const {status, ...p} = props;
   return (
     <StatusWrapper>
-      <CircleIndicator size={6} color={theme[colors[status]]} />
+      <CircleIndicator size={6} color={theme[COLORS[status]]} />
       <div {...p}>{`${t(status)}`}</div>
     </StatusWrapper>
   );
 })`
-  color: ${(props: StatusProps) => theme[colors[props.status]]};
+  color: ${(props: StatusProps) => theme[COLORS[props.status]]};
```

Comment:
Do we do this 'functional component' inside the argument for a styled component elsewhere? It seems a bit confusing rather than separating them out:

```js
const BaseIntegrationSatus = (props: StatusProps) => {
  return (
    <StatusWrapper>
     //...
```
```js
const IntegrationStatus = styled(BaseIntegrationStatus)`
  margin: ${space(0.5)};
  //...
```

Reply:
@leeandher line 51 of `app/views/organizationIntegrations/integrationItem.tsx`. It seems like a fairly common pattern.

Reply:
Good catch @leeandher. @NisanthanNanthakumar You misunderstood the comment.

```
const IntegrationStatus = styled((props: StatusProps) => {})'...'
```

should be rewritten to 

```
const MyComponent = () => { ... }
const IntegrationStatus = styled<StatusProps>(MyComponent)'...'
```

---

Additionally, you should not be asserting the type of the prop on Line 30.

This is not great.
`${(props: StatusProps) => theme[colors[props.status]]}`

This is the better way, you should let IntelliSense do its thing.
`${(props) => theme[colors[props.status]]}`

This is how we do it at Sentry
`${p => theme[colors[p.status]]}`

Reply:
@leedongwei interesting, Is there a good heuristic to follow when deciding whether to explicitly type or use intellisense?

Reply:
If your types are strict and correct, you (usually) won't need to explicitly type. There are very few edge cases where TS struggle to infer the types so you'll have to help it along in those cases.

If you always have to explicitly type frequently, it's a symptom that the underlying types are off.